R/multitables.R

```diff
@@ -99,14 +155,20 @@ newMultitable <- function (formula, data, name, ...) {
         name <- formulaRHS(formula)
     }
 
+    u <- .newMultitable(template, shojiURL(data, "catalogs", "multitables"), name, ...)
+    
+    invisible(Multitable(crGET(u)))
+}
+
+## Internal function to do the POSTing, both in [[ and in newMultitable
+.newMultitable <- function (template, multiable_url, name, ...) {
     payload <- wrapEntity(
         name=name,
         template=lapply(template$dimensions,
-            function (x) list(query=x, variable=findVariableReferences(x)))
+                        function (x) list(query=x, variable=findVariableReferences(x)))
     )
     ## TODO: remove "variable" from that--no longer required?
```

Comment:
Can you do this TODO (and also remove it from where I see it copied above)? Speaking of copying, since the same logic is in two places, might it merit being factored out to a function that is reused?